pinot-query-planner/src/test/resources/queries/GroupByPlans.json

```diff
@@ -249,6 +249,39 @@
           "\n              LogicalTableScan(table=[[default, a]])",
           "\n"
         ]
+      },
```

Comment:
Is this PR meant to enable trimming within (minSegmentTrim) and across segments (minServerTrim) ?
If so - It'd be good to mention that :
- the former is disabled by default  and requires using query option (or cluster setting)
- the latter is enabled by default but probably haven't been applied so far

Apart from manual hinting I think we could propagate limit and order by details to leaf plans and enable both types of trimming if: 
- order by is based on group by key(s) only (not aggregates)
- there is no HAVING clause

Reply:
Leaf stage has both segment trim and server trim. Will add some javadoc about the behavior, and we should consider adding them also as v2 hint. Right now the behavior relies on the v1 query options.

Agree on automating it when the result is guaranteed to be accurate. This will also require some v1 engine change, so will do it separately.